map/place_page_info.cpp

```diff
@@ -32,7 +32,7 @@ bool Info::ShouldShowAddPlace() const
   return !(IsFeature() && isPointOrBuilding);
 }
 
-void Info::SetFromFeatureType(FeatureType & ft)
+void Info::SetFromFeatureType(FeatureType & ft, Framework const * framework)
```

Comment:
This function is getting a bit complex, I would recommend splitting it into smaller functions (one for buildings, one for sidewalks, etc.).

Using free functions would allow to easily see side effects of each function, e.g
```cpp
if (IsBookmark())
  std::tie(m_uiTitle, m_uiSecondaryTitle) = handleBookmark(/* ... */);
else if (IsBuilding())
  std::tie(m_uiTitle, m_uiAddress) = handleBuilding(/* ... */);
else if(IsSidewalk())
  std::tie(m_uiTitle, m_uiAddress) = handleSidewalk(/* ... */);
```

Reply:
Is std::tie needed with C++17 and above?

Will `auto const & [m_uiTitle, m_uiSecondaryTitle] = handleBookmark(/* ... */);` work?

Reply:
No, because `m_uiTitle` and `m_uiSecondaryTitle` are already declared (as class members in this example).

Reply:
Right, we're working with local variables 🤦‍♂️